packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -321,6 +457,55 @@ class IconButton extends StatelessWidget {
     final BoxConstraints adjustedConstraints = effectiveVisualDensity.effectiveConstraints(unadjustedConstraints);
     final double effectiveIconSize = iconSize ?? IconTheme.of(context).size ?? 24.0;
 
+    if (theme.useMaterial3) {
+      final Size? minSize = constraints == null
+          ? null
+          : Size(constraints!.minWidth, constraints!.minHeight);
+      final Size? maxSize = constraints == null
+          ? null
+          : Size(constraints!.maxWidth, constraints!.maxHeight);
```

Comment:
Hello @QuncCccccc , quick question
Why are you calculating new size, shouldn't we be using **adjustedConstraints** instead ?
This results in non-square icon buttons which depends on the icon itself causing weird behavior.
![image](https://user-images.githubusercontent.com/76812169/195950728-72429aa5-0574-4e47-8c95-7f0fe3fe00af.png)